src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -62,6 +64,30 @@ public abstract partial class WebRequestPSCmdlet : PSCmdlet
         #region Authorization and Credentials
 
         /// <summary>
+        /// gets or sets the AllowUnencryptedAuthorization property
+        /// </summary>
+        [Parameter]
+        public virtual SwitchParameter AllowUnencryptedAuthorization { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Authorization property used to determin the Authorization method for the web session.!--
+        /// </summary>
+        [Parameter]
+        [ValidateSet("Basic","Bearer","OAuth")]
+        public virtual string Authorization 
+        { 
+            get
+            {
+                return _authorization;
+            } 
+            set
+            {
+                _authorization = value.ToLower();
+            }
+        }
+        private string _authorization;
```

Comment:
I believe we should use autoproperty and new public enum type for "Basic","Bearer","OAuth" - 'AuthorizationHeaderEnum' .

Reply:
autoproperty? Please let me know what this is.

As for the new Public Enum. I was following suit with `-TransferEncoding` which doesn't have a public enum. I'm open to creating one, but I assume this requires committee approval?

Reply:
I meant https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/auto-implemented-properties
Yes, public API should be approved. I believe it is right place to use it.